actors/market/src/lib.rs

```diff
@@ -231,6 +230,29 @@ impl Actor {
                 }
             };
 
+            // Fetch client's datacap balance and calculate the amount of datacap required for the verified deals.
+            // Drop any verified deals for which the client has insufficient datacap.
+            if deal.proposal.verified_deal {
```

Comment:
This code is decently more complicated than the two blocks following it (https://github.com/filecoin-project/builtin-actors/pull/722/files#diff-98c58c90d2a8f98524c392a7745c8028672c77eb4740c9f823df09c871cb59c4R256-R267) and (https://github.com/filecoin-project/builtin-actors/pull/722/files#diff-98c58c90d2a8f98524c392a7745c8028672c77eb4740c9f823df09c871cb59c4R256-R267)  but it does essentially the same thing.  

If you restructure this to match those blocks below and add the all_client_data_cap update in the same place as client lockup update (https://github.com/filecoin-project/builtin-actors/pull/722/files#diff-98c58c90d2a8f98524c392a7745c8028672c77eb4740c9f823df09c871cb59c4R302).  Then this is going to be easier to read and maintain

Reply:
The `all_client_datacap` map stores total and used for each client, but I think it only needs to store one number: the remaining amount (`total - used`).

It's complex to include the lookup inside this loop. It would be more clear to first find all the clients, fetch all their balances, and then just use this mapping while iterating deals. It would require iterating deals a second time. But we do iterate them all again anyway further on when creating allocation requests. Perhaps we can get that all done up here in one iteration, before the one that this comment is inside.